boot-microservices/src/main/java/org/ow2/proactive/boot/microservices/IAMStarter.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.boot.microservices;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.*;
+
+import org.apache.log4j.Logger;
+
+
+public enum IAMStarter {
+
+    //singleton instance of IAMStarter
+    INSTANCE;
+
+    private static final Logger LOGGER = Logger.getLogger(IAMStarter.class);
+
+    private static final String SEPARATOR = File.separator;
+
+    private static final String MICROSERVICE_NAME = "iam.war";
+
+    private static final String READY_MARKER = "Ready to process requests";
+
+    private static final int TIMEOUT = 180;
+
+    private static final String[] JVM_ARGS = { "-Xmx2048M" };
```

Comment:
is it not better to have this configurable ?

Reply:
Yes, this will done properly, when we adopt this way (a separate JVM process) to execute the IAM microservice